src/Scorecard.ts

```diff
@@ -0,0 +1,56 @@
+import {CategoryScores} from "./domain";
+
+const ONES_THROUGH_SIXES_BONUS_SCORE = 63;
+
+export class Scorecard {
+  private chosenCategories: string[]
+  private scores: CategoryScores;
+  private yahtzeeBonus: boolean;
+
+  constructor() {
+    this.yahtzeeBonus = false;
+    this.chosenCategories = []
+    this.scores = {
+      ones: 0,
+      twos: 0,
+      threes: 0,
+      fours: 0,
+      fives: 0,
+      sixes: 0,
+      threeOfAKind: 0,
+      fourOfAKind: 0,
+      fullHouse: 0,
+      smallStraight: 0,
+      largeStraight: 0,
+      yahtzee: 0,
+      chance: 0,
+    }
+  }
+
+  markYahtzeeBonus(): void {
+    this.yahtzeeBonus = true;
+  }
+
+  hasYahtzeeBonus(): boolean {
+    return this.yahtzeeBonus
+  }
+
+  hasOnesThroughSixesBonus(): boolean {
+    return (
+      this.scores.ones +
+      this.scores.twos +
+      this.scores.threes +
+      this.scores.fours +
+      this.scores.fives +
+      this.scores.sixes) >= ONES_THROUGH_SIXES_BONUS_SCORE
+  }
+
+  isAlreadyChosen(category: string): boolean {
```

Comment:
Might be nice to have categories as an enum since there's an enumerated set of them but doesn't make a big difference

Reply:
This current implementation of `isAlreadyChosen` does still work if a bad value for a category is passed in, but making it an enum would make impossible values impossible to input

Reply:
Definitely agree. I actually had that for a moment - I had a `type Category = keyof CategoryScores` and had this function take a Category type. But it led to a bit of annoyance in the Chooser because `Object.keys` returns a list of strings, and for the time being I put that aside.